src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -197,6 +204,15 @@ def schema_expression(data_type: DataType, is_nullable: bool) -> str:
         return "to_geography('POINT(-122.35 37.55)')"
     if isinstance(data_type, GeometryType):
         return "to_geometry('POINT(-122.35 37.55)')"
+    if isinstance(data_type, VectorType):
+        if data_type.element_type == "int":
+            zero = int(0)
+        elif data_type.element_type == "float":
+            zero = float(0)
+        else:
+            raise TypeError(f"Invalid vector element type: {data_type.element_type}")
+        values = [i + zero for i in range(data_type.dimension)]
```

Comment:
Curious: What is this zero doing?

Reply:
This is to ensure we add float elements when creating float vectors (adding `float(0)` results in `1.0, 2.0, 3.0`). This isn't strictly necessary since internally we can create float vectors from integers, but I added it for clarity for debugging etc.